src/NoSqlProvider.ts

```diff
@@ -46,6 +47,7 @@ export interface DbIndex {
     countOnly(key: any|any[]): SyncTasks.Promise<number>;
     countRange(keyLowRange: any|any[], keyHighRange: any|any[], lowRangeExclusive?: boolean, highRangeExclusive?: boolean)
         : SyncTasks.Promise<number>;
+    fullTextSearch<T>(searchPhrase: string): SyncTasks.Promise<T[]>;
```

Comment:
Question: Is there overhead in supporting fullTextSearch for indexes that do not need it?